### PR TITLE
[V2]Add Script for metrics markdown table

### PR DESCRIPTION
**Pull request by vvs-personalstash**
## Description of the changes
- Added a script for creating a markdown table to view and compare v1 and v2 metrics using json generated by compare metrics

## Checklist
- *[x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- *[x] I have signed all commits
- [ ] I have added unit tests for the new functionality
- *[x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

**Comment by Wise-Wizard** [comment]
In terminal, run `make fmt` and push again.

File: scripts/metrics-md.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import json
```
Comment (yurishkuro) [inline scripts/metrics-md.py:1]: * move this into script/utils
* please document how to use that
* include generated MD file(s) - you can put then under cmd/jaeger/docs/migration

File: cmd/jaeger/docs/migration/all-in-one-metrics.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+| Metric Name | Inner Parameters |
+|-------------|------------------|
+| jaeger_query_latency | operation, result |
+| jaeger_query_responses | operation |
+| jaeger_query_requests_total | operation, result |
+### V1 Only Metrics
+
+| Metric Name | Inner Parameters |
+|-------------|------------------|
+| go_gc_duration_seconds |  |
```
Comment (yurishkuro) [inline cmd/jaeger/docs/migration/all-in-one-metrics.md:12]: @Wise-Wizard we should investigate how these standard Go metrics are being generated in v1 and decide how we can enable them in v2.

File: cmd/jaeger/docs/migration/all-in-one-metrics.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
+### Common Metrics
+
+| Metric Name | Inner Parameters |
```
Comment (yurishkuro) [inline cmd/jaeger/docs/migration/all-in-one-metrics.md:3]: > Inner Parameters

The official term in Prometheus is Labels

File: scripts/utils/metrics-md.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        text=common_metrics_table+v1_metrics_table+v2_metrics_table+spans_metrics_table
+        return text
+
+    def convert_dict_to_md(self, output_fn):
+        with open(output_fn, 'w') as writer:
+            writer.writelines(self.mddata)
+        print('Dict successfully converted to md')
+
+# Usage
+fn = ''     #Enter the path of the JSON file generated by compare_metrics.py
```
Comment (yurishkuro) [inline scripts/utils/metrics-md.py:96]: a. let's move compare_metrics.py to utils too.
b. could we just add the logic here to compare_metrics.py ? Maybe as an optional parameter `--out=json|md`

Comment (yurishkuro) [reply scripts/utils/metrics-md.py:96]: 3. I still don't see instructions how to run these things - the compare_metrics.py seems to hardcode the input file names like `./V1_Metrics.json` -- please document how those JSON files should be produced.

File: cmd/jaeger/docs/migration/badger-metrics.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+| process_runtime_heap_alloc_bytes | service_instance_id, service_name, service_version |
+| process_runtime_total_alloc_bytes | service_instance_id, service_name, service_version |
+| process_runtime_total_sys_memory_bytes | service_instance_id, service_name, service_version |
+| process_uptime | service_instance_id, service_name, service_version |
+| processor_batch_batch_send_size | processor, service_instance_id, service_name, service_version |
+| processor_batch_metadata_cardinality | processor, service_instance_id, service_name, service_version |
+| processor_batch_timeout_trigger_send | processor, service_instance_id, service_name, service_version |
+| receiver_accepted_spans | receiver, service_instance_id, service_name, service_version, transport |
+| receiver_refused_spans | receiver, service_instance_id, service_name, service_version, transport |
+| target_info | service_instance_id, service_name, service_version |
```
Comment (yurishkuro) [inline cmd/jaeger/docs/migration/badger-metrics.md:101]: arguably this is equivalent to `jaeger_build_info`

File: cmd/jaeger/docs/migration/badger-metrics.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
+### Common Metrics
+
+| Metric Name | Inner Parameters |
+|-------------|------------------|
+| jaeger_badger_compaction_current_num_lsm |  |
```
Comment (yurishkuro) [inline cmd/jaeger/docs/migration/badger-metrics.md:5]: the storage-specific files will have large overlap with all-in-one, I recommend filtering out only the storage-related metrics.

File: cmd/jaeger/docs/migration/all-in-one-metrics.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
+### Common Metrics
```
Comment (yurishkuro) [inline cmd/jaeger/docs/migration/all-in-one-metrics.md:1]: could we just have a single table with a format similar to `### Equivalent Metrics`? When one side is missing, put n/a

File: cmd/jaeger/docs/migration/all-in-one-metrics.md (hunk 2 above)
Comment (yurishkuro) [reply cmd/jaeger/docs/migration/all-in-one-metrics.md:12]: this is what we might want: https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/runtime/runtime.go

ideally Collector would expose those automatically, but it's not implemented currently -- https://github.com/open-telemetry/opentelemetry-collector/issues/2155

File: scripts/utils/metrics-md.py (hunk 4 above)
Comment (vvs-personalstash) [reply scripts/utils/metrics-md.py:96]: ok should i create a separate documentation md file for this in scripts/utils or should i just add the instruction as comments

Comment (yurishkuro) [reply scripts/utils/metrics-md.py:96]: in comments is fine, this isn't something we'd use often

**Comment by vvs-personalstash** [comment]
@yurishkuro I have made changes as per your reviews,kindly review if there are any changes or errors i may have not noticed

**Pull request merged**